warehouse/query-core/src/main/java/datawave/query/jexl/functions/QueryFunctionsDescriptor.java

```diff
@@ -126,24 +129,74 @@ public void addFilters(AttributeFactory attributeFactory, Map<String,EventDataQu
 
         @Override
         public Set<String> fieldsForNormalization(MetadataHelper helper, Set<String> datatypeFilter, int arg) {
-            // Do not normalize fields for the includeText function.
-            if (!name.equalsIgnoreCase(INCLUDE_TEXT)) {
-                // All other functions use the fields in the first argument for normalization.
-                if (arg > 0) {
-                    return fields(helper, datatypeFilter);
-                }
+            if (name.equalsIgnoreCase(QueryFunctions.INCLUDE_TEXT)) {
```

Comment:
Read the function description.  The arg argument is to be used to denote argument we are to be returning the fields for normalization.  So for example if the first argument denotes the name of the fields, then for arg=0, this should return the empty set.

Reply:
Got it, I'll revert that change. Do any additional functions need to delegate to the fields method besides `includeText` (such as `matchRegex`) ?

Reply:
So the idea was that the "fields" being returned are the the fields for which thier normalizers should be applied to the argument in question.  Originally all of the methods being handled by the descriptor essentially would need to normalize all arguments past the first one using all of the fields, except for include_text which was explicitly made to bypass any normalization.
So to answer your question, look at the arguments for all methods being handled by the QueryFunctions and see if that holds.  If not then we may need to adjust something.

Reply:
After some poking around I don't believe matchRegex needs to be added to the exclusion list.